### PR TITLE
Sergey/eng 302 reshape cli commands to start hooks

### PR DESCRIPTION
**Pull request by spetrunin**
#### Checklist

- [ ] run `make test`
- [ ] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Code of conduct](https://github.com/wundergraph/wundergraph/blob/main/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: cli/runners/server.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	ServerHost        string
+	NodeAddr          string
+	ServerScriptFile  string
+}
+
+func NewServerRunner(log abstractlogger.Logger, cfg *ServerRunConfig) *scriptrunner.ScriptRunner {
+	hooksEnv := []string{
+		"START_HOOKS_SERVER=true",
+		fmt.Sprintf("WG_ABS_DIR=%s", cfg.WunderGraphDirAbs),
+		fmt.Sprintf("HOOKS_TOKEN=%s", cfg.HooksJWT),
```
Comment (spetrunin) [inline cli/runners/server.go:26]: prefix with WG

File: cli/commands/node.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+		}
+		var wunderNodeConfig wgpb.WunderNodeConfig
+		err = json.Unmarshal(data, &wunderNodeConfig)
+		if err != nil {
+			log.Fatal("Failed to unmarshal", abstractlogger.String("filePath", configFile), abstractlogger.Error(err))
+			return err
+		}
+
+		var settings NodeStartSettings
+		if err := envconfig.Process("", &settings); err != nil {
```
Comment (spetrunin) [inline cli/commands/node.go:70]: add "WG" prefix

File: cli/commands/server.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+		}
+
+		serverScriptFile := path.Join("generated", "bundle", "server.js")
+		serverExecutablePath := path.Join(entryPoints.WunderGraphDirAbs, "generated", "bundle", "server.js")
+		if !files.FileExists(serverExecutablePath) {
+			return fmt.Errorf(`hooks server build artifact "%s" not found. Please use --exclude-server to disable the server`, path.Join(wundergraphDir, serverScriptFile))
+		}
+
+		var settings ServerStartSettings
+		if err := envconfig.Process("", &settings); err != nil {
```
Comment (spetrunin) [inline cli/commands/server.go:51]: add "WG" prefix

File: cli/commands/node.go [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+			},
+		}
+
+		n := node.New(ctx, BuildInfo, cfg, log)
+
+		go func() {
+			err := n.StartBlocking(
+				node.WithStaticWunderNodeConfig(wunderNodeConfig),
+				node.WithHooksSecret(settings.Secret),
+				node.WithHooksServerUrl(settings.ServerUrl),
```
Comment (spetrunin) [inline cli/commands/node.go:92]: q: should we use this in start command?

File: packages/sdk/src/env/index.ts [hunk 5]
```diff
@@ -1,5 +1,11 @@
-export const listenAddr = 'WG_LISTEN_ADDR' in process.env ? process.env['WG_LISTEN_ADDR']! : 'localhost:9991';
-export const listenAddrHttp = 'WG_LISTEN_ADDR' in process.env ? `http://${listenAddr}` : 'http://localhost:9991';
-export const middlewarePortString: string =
-	'WG_MIDDLEWARE_PORT' in process.env ? process.env['WG_MIDDLEWARE_PORT']! : '9992';
-export const middlewarePort = parseInt(middlewarePortString, 10);
+// node config
```
Comment (spetrunin) [inline packages/sdk/src/env/index.ts:1]: make sure to add docs/migrations for env variable changes
cause we removed some variables from exports and renamed other ones

File: cli/commands/generate.go [hunk 6]
```diff
@@ -145,6 +147,7 @@ func init() {
 	generateCmd.Flags().BoolVarP(&generateAndPublish, "publish", "p", false, "publish the generated API immediately")
 	generateCmd.Flags().StringVar(&listenAddr, "listen-addr", "localhost:9991", "listen_addr is the host:port combination, WunderGraph should listen on.")
 	generateCmd.Flags().IntVar(&middlewareListenPort, "middleware-listen-port", 9992, "middleware-listen-port is the port which the WunderGraph middleware will bind to")
+	generateCmd.Flags().StringVar(&middlewareHost, "middleware-host", "127.0.0.1", "middleware-host is the host which the WunderGraph middleware will bind to")
```
Comment (StarpTech) [inline cli/commands/generate.go:150]: Please rename `middleware` to `server` and add this change to the migration guide.

File: cli/commands/node.go [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+
+var nodeCmd = &cobra.Command{
+	Use:   "node",
+	Short: "Subcommand to work with WunderGraph node",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return nil
+	},
+}
+
+type NodeStartSettings struct {
```
Comment (StarpTech) [inline cli/commands/node.go:29]: Ideally, we can handle ENV and cli flags with https://github.com/spf13/viper#working-with-environment-variables

File: cli/commands/node.go [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+	Secret    []byte `envconfig:"SECRET" required:"true"`
+	ServerUrl string `envconfig:"SERVER_URL" required:"true"`
+}
+
+var nodeStartCmd = &cobra.Command{
+	Use:   "start",
+	Short: "Start runs WunderGraph Node in production mode",
+	Long: `
+		Example usage:
+			SECRET=secret SERVER_URL=127.0.0.1 SERVER_PORT=9993 NODE_URL=127.0.0.1:9991 wunderctl node start
```
Comment (StarpTech) [inline cli/commands/node.go:40]: They should be prefixed with `WG_`

File: cli/commands/node.go (hunk 4 above)
Comment (StarpTech) [reply cli/commands/node.go:92]: Yes :+1: much better than working with a channel

File: cli/commands/server.go [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+
+	"github.com/kelseyhightower/envconfig"
+	"github.com/spf13/cobra"
+
+	"github.com/wundergraph/wundergraph/cli/runners"
+	"github.com/wundergraph/wundergraph/pkg/apihandler"
+	"github.com/wundergraph/wundergraph/pkg/files"
+)
+
+type ServerStartSettings struct {
```
Comment (StarpTech) [inline cli/commands/server.go:16]: Same for here. We should find a way to configure CLI with env and flags.

File: cli/commands/up.go [hunk 10]
```diff
@@ -324,6 +321,7 @@ func init() {
 	rootCmd.AddCommand(upCmd)
 	upCmd.Flags().StringVar(&listenAddr, "listen-addr", "localhost:9991", "listen_addr is the host:port combination, WunderGraph should listen on.")
 	upCmd.Flags().IntVar(&middlewareListenPort, "middleware-listen-port", 9992, "middleware-listen-port is the port which the WunderGraph middleware will bind to")
+	upCmd.Flags().StringVar(&middlewareHost, "middleware-host", "127.0.0.1", "middleware-host is the host which the WunderGraph middleware will bind to")
```
Comment (StarpTech) [inline cli/commands/up.go:324]: Rename to `server`

**Comment by spetrunin** [comment]
closed in favor of https://github.com/wundergraph/wundergraph/pull/151

**Pull request closed without merging**